rfcs/039-table-data-quality-checks.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: table_data_quality_checks
+- Start Date: 2021-07-16
+- RFC PR: [amundsen-io/rfcs#XX](https://github.com/amundsen-io/rfcs/pull/XX)
+# Table Data Quality Checks
+
+## Summary
+This RFC will define a generic interface for accessing data quality checks in Amundsen. If available, we aim to expose
+the status of these data quality checks in the Amundsen Table Details page.
+
+
+## Terms
+- Data Quality Check - Like unit tests for code, data quality checks are assertions that try to guarantee certain 
+  characteristics of data. For example a data quality check could be as simple as asserting that a column is not null,
+  or implement some complex custom logic.
+
+
+## Motivation
+While Amundsen is not currently building unit tests for data, we can empower our users by allowing integrations 
+with a generic external data quality service. Exposing the status of data quality checks in Amundsen can give users 
+an at-a-glance summary of the table's quality. This can help a user to quickly decide if they want to use this table.
+
+## Product Details
+Users will see a new section on the left panel of the Table Details page titled `Data Quality Checks`. There will be
+a message and icon for the current status of these checks, if they exist. A passing table might say 
+`✅ 10 of 10 checks passed` and a failing table might say `❌ 3 of 10 checks failed`. Additionally, the user may see 
+the last run timestamp, and a link to the external data quality service, if available.
+
+## UX Explanation
+A new section will be added to the left side of the Table Details Page.
+
+![Data Quality Preview](../assets/039/data_quality.png)
+
+## Technical Details
+Table Data Quality Checks will be implemented as an optional feature on Amundsen Frontend. This feature will be hidden 
+behind a config flag and will use a generic client much like the existing data preview client. The generic client should
+implement the following interface: 
+
+```
+def get_table_quality_checks(self, *, table_key: str):
+
+Response:  
+{
+  num_checks_total: Int;
+  num_checks_success: Int;
+  num_checks_failed: Int;
+  external_url: Optional[Str];
+  last_run_timestamp: Optional[Int];
+}
+```
+
+## Drawbacks
+This can be seen as a somewhat shallow integration that doesn't allow for some deeper integrations. Granted that we are
+trying to support a generic model, it's difficult to support a deeper integration given that we are not endorsing a 
+data quality service provider.
+
+## Alternatives
+There are several alternatives: 
+  - Ingest data quality checks as part of the databuilder and display the status as part of the table metadata API. 
+The displayed status in this case may be stale.
+  - We could build data quality checks as an Amundsen service and provide a much deeper level of service, but this 
```

Comment:
I'm similarly negative on this: integration is the way here

Reply:
+1, we should link dq service that is hosted externally instead of putting DQ inside amundsen which is more for discovery purpose.

Reply:
+1 for OPs statement.

In regards to:

> we should link dq service that is hosted externally instead of putting DQ inside amundsen which is more for discovery purpose.

There are some dq services that are batch processes which produce a static output in JSON, html, etc.  Are you suggesting that Amundsen should not coordinate and execute the dq tests _or_ is this comment saying that Amundsen should not ingest the output from the tests that provide table/column level quality results?

Reply:
I think we're saying Amundsen should not be the entity responsible for registering and running the DQ tests. We should be integrating with an external DQ check service that can provide this information.